bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/esp32/WlanThermoEsp32Handler.java

```diff
@@ -0,0 +1,222 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.wlanthermo.internal.api.esp32;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.concurrent.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Authentication;
+import org.eclipse.jetty.client.api.AuthenticationStore;
+import org.eclipse.jetty.client.util.DigestAuthentication;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.openhab.binding.wlanthermo.internal.WlanThermoBindingConstants;
+import org.openhab.binding.wlanthermo.internal.WlanThermoExtendedConfiguration;
+import org.openhab.binding.wlanthermo.internal.api.esp32.dto.data.Data;
+import org.openhab.binding.wlanthermo.internal.api.esp32.dto.settings.Settings;
+import org.openhab.core.thing.*;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link WlanThermoEsp32Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Christian Schlipp - Initial contribution
+ */
+@NonNullByDefault
+public class WlanThermoEsp32Handler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(WlanThermoEsp32Handler.class);
+
+    private WlanThermoExtendedConfiguration config = new WlanThermoExtendedConfiguration();
+    private WlanThermoEsp32CommandHandler wlanThermoEsp32CommandHandler = new WlanThermoEsp32CommandHandler();
+    private final HttpClient httpClient;
+    private @Nullable ScheduledFuture<?> pollingScheduler;
+    private final Gson gson = new Gson();
+    private @Nullable Data data = new Data();
+    private @Nullable Settings settings = new Settings();
+
+    public WlanThermoEsp32Handler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(WlanThermoExtendedConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        try {
+            if (!config.getUsername().isEmpty() && !config.getPassword().isEmpty()) {
+                AuthenticationStore authStore = httpClient.getAuthenticationStore();
+                authStore.addAuthentication(new DigestAuthentication(config.getUri(), Authentication.ANY_REALM,
+                        config.getUsername(), config.getPassword()));
+            }
+            pollingScheduler = scheduler.schedule(this::checkConnection, config.getPollingInterval(), TimeUnit.SECONDS);
```

Comment:
Can you merge the `checkConnection` and `update` methods into a single method so that you only require a single period task to be set during `initialize`? Things can go wrong if you are assigning the `pollingScheduler` field in other threads outside of the `initialize` and `dispose` methods.